### PR TITLE
Introduce rpc.Params

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
**What changed?**
Introduced `rpc.Params` structure that contains required parameters to initialize `rpc.Factory`.
This decouples Factory from using config directly. It is beneficial for 2 reasons:
- Allows unit test config part easier
- Allows reusing the same `rpc.Factory` in integration tests. Before we had a separate implementation there, with minor differences. Now we can pass those differences via Params.

Also refactored service config retrieval into a function. This is now used in couple of places.

**Why?**
To reduce code duplication and increase test coverage.

**How did you test it?**
Additional unit tests for refactored code.
Existing tests.
Starting server locally with removed config section.

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: cmd/server/cadence/cadence.go [hunk 1]
```diff
@@ -82,9 +82,6 @@ func startHandler(c *cli.Context) {
 	sigc := make(chan os.Signal, 1)
 	signal.Notify(sigc, syscall.SIGTERM)
 	for _, svc := range services {
-		if _, ok := cfg.Services[svc]; !ok {
```
Comment (vytautas-karpavicius) [inline cmd/server/cadence/cadence.go:85]: No need to check here. This is checked down the line in `server.startService` when config is actually needed.

File: cmd/server/cadence/server.go [hunk 2]
```diff
@@ -70,9 +70,6 @@ func newServer(service string, cfg *config.Config) common.Daemon {
 
 // Start starts the server
 func (s *server) Start() {
-	if _, ok := s.cfg.Services[s.name]; !ok {
```
Comment (vytautas-karpavicius) [inline cmd/server/cadence/server.go:73]: No need to check here. This is checked down the line in `server.startService` when config is actually needed.

**Review by mantas-sidlauskas — APPROVED** [review]
left one minor comment

File: common/rpc/grpc.go [hunk 3]
```diff
@@ -29,7 +29,13 @@ import (
 	"github.com/uber/cadence/common/service"
 )
 
-type GRPCPorts map[string]int
+type (
+	GRPCPortResolver interface {
+		GetGRPCAddress(service, hostAddress string) (string, error)
+	}
```
Comment (mantas-sidlauskas) [inline common/rpc/grpc.go:35]: Is this interface really needed?

If so, I would make it more general, as it does nothing specific to `grpc`

Comment (vytautas-karpavicius) [reply common/rpc/grpc.go]: The interface is needed, as integration tests pass different implementation.
But I agree that the name could be more general. Updated it.

**Pull request merged**